data-store/chapter.md

```diff
@@ -182,9 +129,29 @@ from the logical structure of the data
 from the contents of the key/value store
 (the association of key "a" to value "foo"; the public API).
 
+[TODO: Sentence or two about the common pattern of separating logical from
+physical, and why it's used in many database designs.]
+
+Discovering the design
+----------------------
+
+Most of the chapters in this book describe how a program was built from
+inception to completion. However, that is not how most of us interact with the
+code we're working on. We are most often discovering code that was written by
+others, and figuring out how to modify or extend it to do something
+differently.
```

Comment:
"We most often discover code that was written by others, and figure out how to modify or extend it to do something different."